File: MuscleHamster/Utilities/FeatureFlags.swift
import Foundation

// MARK: - Feature Flags
// Controls which features are enabled in the simplified MVP.
// Set flags to false to defer features to v1.1+

enum FeatureFlags {

    // MARK: - Deferred Features (OFF for Simplified MVP)

    /// Social features: friends, friend streaks, nudges, blocking, privacy controls
    static let socialFeatures = false

    /// Full workout library with browse, filter, recommendations, workout player
    static let workoutLibrary = false

    /// Rest day check-in micro-tasks (pet hamster, give treat, log activity)
    static let restDayCheckIn = false

    /// Complex onboarding questions (fitness level, goals, schedule, etc.)
    static let complexOnboarding = false

    /// Transaction history view in settings
    static let transactionHistory = false

    /// Advanced notifications (streak at risk, custom scheduling)
    static let advancedNotifications = false

    /// Audio system (SFX, music, volume controls)
    static let audioSystem = false

    /// Tips rotation on home screen
    static let tipsRotation = false

    /// Rarity system for shop items
    static let raritySystem = false

    /// Workout feedback (loved/liked/not for me)
    static let workoutFeedback = false

    // MARK: - Core Features (ON for Simplified MVP)

    /// Daily exercise check-in (the ONE action)
    static let dailyExerciseCheckIn = true

    /// Hamster display and customization
    static let hamsterCustomization = true

    /// Streak tracking and streak freeze
    static let streakSystem = true

    /// Points economy and shop
    static let pointsAndShop = true

    /// Growth milestones and celebrations (hamster aging from baby to mature)
    /// When false, hamster is always one stage (no growth)
    static let growthMilestones = false

    /// Basic notifications (daily reminder on/off)
    static let basicNotifications = true

    // MARK: - Navigation

    /// Use tab bar navigation (false = hub navigation from Home)
    static let tabBarNavigation = false

    /// Number of shop items to show (simplified = 12, full = 24)
    static let simplifiedShopItemCount = 12

    // MARK: - Onboarding

    /// Number of onboarding steps (simplified = 2: name hamster, meet hamster)
    /// Full = 8 steps including all fitness questions
    static let simplifiedOnboarding = true

    // MARK: - Hamster States

    /// Use simplified hamster states (happy/hungry only)
    /// Full = happy, chillin, hungry, excited, proud
    static let simplifiedHamsterStates = true
}

// MARK: - Feature Flag Helpers

extension FeatureFlags {

    /// Check if any social features should be shown
    static var showSocialTab: Bool {
        socialFeatures
    }

    /// Check if workouts tab should be shown
    static var showWorkoutsTab: Bool {
        workoutLibrary
    }

    /// Check if rest day option should be shown on home
    static var showRestDayOption: Bool {
        restDayCheckIn
    }

    /// Check if tips should rotate on home screen
    static var showTipsCard: Bool {
        tipsRotation
    }

    /// Check if audio settings should be shown
    static var showAudioSettings: Bool {
        audioSystem
    }

    /// Check if privacy settings should be shown (only needed with social)
    static var showPrivacySettings: Bool {
        socialFeatures
    }

    /// Check if workout schedule settings should be shown
    static var showWorkoutScheduleSettings: Bool {
        workoutLibrary
    }

    /// Check if profile editing should show fitness questions
    static var showFitnessProfileEditing: Bool {
        complexOnboarding
    }
}

File: MuscleHamsterExpo/src/config/FeatureFlags.js
// Feature Flags - Simplified MVP
// Controls which features are enabled. Set to false to defer features to v1.1+

export const FeatureFlags = {
  // DEFERRED FEATURES (OFF for Simplified MVP)

  // Social features: friends, friend streaks, nudges, blocking, privacy controls
  socialFeatures: false,

  // Full workout library with browse, filter, recommendations, workout player
  workoutLibrary: true,

  // Rest day check-in micro-tasks (pet hamster, give treat, log activity)
  restDayCheckIn: false,

  // Complex onboarding questions (fitness level, goals, schedule, etc.)
  complexOnboarding: false,

  // Transaction history view in settings
  transactionHistory: false,

  // Advanced notifications (streak at risk, custom scheduling)
  advancedNotifications: false,

  // Audio system (SFX, music, volume controls)
  audioSystem: false,

  // Tips rotation on home screen
  tipsRotation: false,

  // Rarity system for shop items
  raritySystem: false,

  // Workout feedback (loved/liked/not for me)
  workoutFeedback: false,

  // Growth milestones (hamster aging from baby to mature)
  // When false, hamster is always one stage (adult)
  growthMilestones: false,

  // CORE FEATURES (ON for Simplified MVP)

  // Daily exercise check-in (the ONE action)
  dailyExerciseCheckIn: true,

  // Hamster display and customization
  hamsterCustomization: true,

  // Streak tracking and streak freeze
  streakSystem: true,

  // Points economy and shop
  pointsAndShop: true,

  // Basic notifications (daily reminder on/off)
  basicNotifications: true,

  // NAVIGATION

  // Use tab bar navigation (false = hub navigation from Home)
  tabBarNavigation: true,

  // Simplified onboarding (2 steps: age gate + name hamster)
  simplifiedOnboarding: true,

  // Simplified hamster states (happy/hungry only)
  simplifiedHamsterStates: true,
};

// Helper functions
export const showSocialTab = () => FeatureFlags.socialFeatures;
export const showWorkoutsTab = () => FeatureFlags.workoutLibrary;
export const showRestDayOption = () => FeatureFlags.restDayCheckIn;
export const showTipsCard = () => FeatureFlags.tipsRotation;
export const showAudioSettings = () => FeatureFlags.audioSystem;
export const showPrivacySettings = () => FeatureFlags.socialFeatures;
export const showTransactionHistory = () => FeatureFlags.transactionHistory;
export const showGrowthMilestones = () => FeatureFlags.growthMilestones;

export default FeatureFlags;
